### PR TITLE
Pet keybinds!

### PR DESCRIPTION
**Pull request by SubAt0m1c**
+ some other minor stuff (Changed a setting in GoldorTimer to duelsetting, added an option to change color based on distance in itemhighlight)

### CODE REVIEW

File: odinmain/src/main/kotlin/me/odinmain/commands/impl/PetCommand.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import me.odinmain.features.impl.skyblock.PetKeybinds.petList
+import me.odinmain.utils.skyblock.itemID
+import me.odinmain.utils.skyblock.modMessage
+import me.odinmain.utils.skyblock.uuid
+
+val petCommand = commodore("petkeys") {
+
+    literal("add").runs {
+        val petID = if (mc.thePlayer?.heldItem.itemID == "PET") mc.thePlayer?.heldItem.uuid else null
+        if (petID == null) return@runs modMessage("You can only add pets to the pet list!")
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/commands/impl/PetCommand.kt:15]: just do this in the assignment of the variable

File: odinmain/src/main/kotlin/me/odinmain/commands/impl/PetCommand.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        if (petID in petList) return@runs modMessage("This pet is already in the list!")
+
+        petList.add(petID)
+        modMessage("Added this pet to the pet list in position ${petList.indexOf(petID) +1}!")
+        Config.save()
+    }
+
+    literal("petpos").runs {
+        val petID = if (mc.thePlayer?.heldItem.itemID == "PET") mc.thePlayer?.heldItem.uuid else null
+        if (petID == null) return@runs modMessage("This is not a pet!")
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/commands/impl/PetCommand.kt:26]: again

File: odinmain/src/main/kotlin/me/odinmain/commands/impl/PetCommand.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    literal("clear").runs {
+        petList.clear()
+        modMessage("Cleared the pet list!")
+        Config.save()
+    }
+
+    literal("list").runs {
+        if (petList.size == 0) return@runs modMessage("pet list is empty")
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/commands/impl/PetCommand.kt:48]: big P

File: odinmain/src/main/kotlin/me/odinmain/features/impl/skyblock/ItemsHighlight.kt [hunk 4]
```diff
@@ -29,6 +30,11 @@ object ItemsHighlight : Module(
     }
 
     private fun getEntityOutlineColor(entity: EntityItem): Color {
-        return getRarity(entity.entityItem.lore)?.color ?: Color.WHITE
+        return if (!colorStyle) getRarity(entity.entityItem.lore)?.color ?: Color.WHITE
+        else if (entity.getDistanceToEntity(mc.thePlayer) <= 3.5) {
+            Color.GREEN
+        } else {
+            Color.RED
+        }
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/skyblock/ItemsHighlight.kt:38]: do this in one line

File: odinmain/src/main/kotlin/me/odinmain/features/impl/skyblock/PetKeybinds.kt [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+    val petList: MutableList<String> by ListSetting("List", mutableListOf())
+
+    @SubscribeEvent
+    fun checkKeybinds(event: GuiScreenEvent.KeyboardInputEvent.Pre) {
+        val chest = (event.gui as? GuiChest)?.inventorySlots ?: return
+        if (chest !is ContainerChest) return
+
+        val matchResult = Regex("Pets \\((\\d)/(\\d)\\)").find(chest.name) ?: return
+        val (current, total) = matchResult.destructured
```
Comment (freebonsai) [inline odinmain/src/main/kotlin/me/odinmain/features/impl/skyblock/PetKeybinds.kt:50]: you sure this works, i would think the first part of the destructured is just the matched string, but if it works its fine

**Review by freebonsai — CHANGES_REQUESTED** [review]

File: odinmain/src/main/kotlin/me/odinmain/features/impl/skyblock/PetKeybinds.kt (hunk 5 above)
Comment (SubAt0m1c) [reply odinmain/src/main/kotlin/me/odinmain/features/impl/skyblock/PetKeybinds.kt:50]: worked when i tested it

**Pull request merged**